File: src/Single Number/Single Number.cpp
/**
 * sol1: map 
 * sol2: sort + iteration compare
 * sol3: xor, the "GOD" solution
 */

class Solution {
	public:
		int singleNumber(int A[], int n) {
			while(--n) A[n-1] ^= A[n];
			return A[0];
		}
};

File: src/Single Number/Single Number.py
class Solution:
	# @param A, a list of integer
	# @return an integer
	def singleNumber(self, A):
		for i in xrange(len(A)-1, -1, -1):
			if not i: return A[0]
			else: A[i-1] ^= A[i]
